Source/Extensions/Blazorise.SignaturePad/SignaturePad.razor.cs

```diff
@@ -0,0 +1,283 @@
+                    ImageQuality = new { Changed = imageQualityChanged, Value = paramImageQuality },
+                    IncludeImageBackgroundColor = new { Changed = includeImageBackgroundColorChanged, Value = paramIncludeImageBackgroundColor },
+                    ReadOnly = new { Changed = readOnlyChanged, Value = paramReadOnly },
+                } ) );
+            }
+        }
+
+        await base.SetParametersAsync( parameters );
+    }
+
+    /// <inheritdoc/>
+    protected override async Task OnAfterRenderAsync( bool firstRender )
+    {
+        if ( firstRender )
+        {
+            DotNetObjectRef ??= DotNetObjectReference.Create( this );
+
+            JSModule = new JSSignaturePadModule( JSRuntime, VersionProvider );
+
+            await JSModule.Initialize( DotNetObjectRef, ElementRef, ElementId, new
+            {
+                Value,
+                DotSize,
+                MinLineWidth,
+                MaxLineWidth,
+                Throttle,
+                MinDistance,
+                BackgroundColor,
+                PenColor,
+                VelocityFilterWeight,
+                ImageType = ToImageTypeString( ImageType ),
+                ImageQuality,
+                IncludeImageBackgroundColor,
+                ReadOnly,
+            } );
+        }
+
+        await base.OnAfterRenderAsync( firstRender );
+    }
+
+    /// <inheritdoc/>
+    protected override async ValueTask DisposeAsync( bool disposing )
+    {
+        if ( disposing && Rendered )
+        {
+            await JSModule.SafeDestroy( ElementRef, ElementId );
+
+            await JSModule.SafeDisposeAsync();
+
+            if ( DotNetObjectRef != null )
+            {
+                DotNetObjectRef.Dispose();
+                DotNetObjectRef = null;
+            }
+        }
+
+        await base.DisposeAsync( disposing );
+    }
+
+    /// <summary>
+    /// Clears the content of a signature canvas.
+    /// </summary>
+    /// <returns>A task that represents the asynchronous operation.</returns>
+    public async Task Clear()
+    {
+        if ( JSModule != null )
+        {
+            await JSModule.Clear( ElementRef, ElementId );
+        }
+    }
+
+    /// <summary>
+    /// This method is called by JavaScript when a stroke has ended in the signature pad. It takes the encoded image
+    /// from the signature pad, converts it to bytes and sets the Value property of the component to the image data.
+    /// It then invokes the ValueChanged and EndStroke events asynchronously to notify any subscribers of the change.
+    /// </summary>
+    [JSInvokable]
+    public async Task NotifyEndStroke( string value )
+    {
+        if ( string.IsNullOrEmpty( value ) )
+            return;
+
+        var valueParts = value.Split( ',' );
+
+        if ( valueParts.Length > 1 )
+        {
+            Value = Convert.FromBase64String( valueParts[1] );
```

Comment:
what's in `valueparts[0]`?

Should we just call `EndStroke` with the original value, in case someone needs it?

Should `EndStroke` always be called even if `valueparts.Length>1` is not true?

`EndStroke` is missing argument?